starlite/app.py

```diff
@@ -243,50 +240,89 @@ def __init__(
             tags: A list of string tags that will be appended to the schema of all route handlers under the application.
             template_config: An instance of [TemplateConfig][starlite.config.TemplateConfig]
         """
-
+        # set pre-initialized values
         self._init = False
         self._registered_routes: Set[BaseRoute] = set()
         self._route_handler_index: Dict[str, HandlerIndex] = {}
         self._static_paths: Set[str] = set()
-        self.after_exception = as_async_callable_list(after_exception) if after_exception else []
-        self.after_shutdown = as_async_callable_list(after_shutdown) if after_shutdown else []
-        self.after_startup = as_async_callable_list(after_startup) if after_startup else []
-        self.allowed_hosts = allowed_hosts
-        self.before_send = as_async_callable_list(before_send) if before_send else []
-        self.before_shutdown = as_async_callable_list(before_shutdown) if before_shutdown else []
-        self.before_startup = as_async_callable_list(before_startup) if before_startup else []
-        self.cache = cache_config.to_cache()
-        self.compression_config = compression_config
-        self.cors_config = cors_config
-        self.csrf_config = csrf_config
-        self.debug = debug
-        self.on_shutdown = on_shutdown or []
-        self.on_startup = on_startup or []
-        self.openapi_config = openapi_config
         self.openapi_schema: Optional["OpenAPI"] = None
         self.plain_routes: Set[str] = set()
-        self.plugins = plugins or []
         self.route_map: RouteMapNode = {}
         self.routes: List[BaseRoute] = []
         self.state = State()
-        self.static_files_config = static_files_config
-        self.template_engine = create_template_engine(template_config)
-        super().__init__(
+
+        # creates app config object from parameters
+        app_conf = AppConfig(
```

Comment:
I'd call this `config` rather than `app_conf` for brevity. 

Reason: I'm personally allergic to abbreviations like this. So I'd go for either  either `app_config` or `config`.

Reply:
`config` it is - have to repeat it a lot so was trying to keep the var size down.